### PR TITLE
add more nginx ingress configuration features

### PR DESCRIPTION
**Pull request by pgvishnuram**
## Description

add more nginx ingress configuration features

## Related Issues

Related astronomer/issues#6389

## Testing

QA effort not required

## Merging

cherry-pick to all valid branches

### CODE REVIEW

File: charts/nginx/values.yaml [hunk 1]
```diff
@@ -28,6 +28,10 @@ replicasDefaultBackend: 2
 maxUnavailable: 25%
 maxUnavailableDefaultBackend: 25%
 
+# -- `minReadySeconds` to avoid killing pods before we are ready
+##
```
Comment (danielhoherd) [inline charts/nginx/values.yaml:32]:
```suggestion
```

File: charts/nginx/values.yaml [hunk 2]
```diff
@@ -85,6 +89,16 @@ electionId: ingress-controller-leader
 # Enables the annotation validation feature
 enableAnnotationValidations: false
 
+# -- This configuration enables Topology Aware Routing feature, used together with service annotation service.kubernetes.io/topology-mode="auto"
+# Defaults to false
```
Comment (danielhoherd) [inline charts/nginx/values.yaml:93]:
```suggestion
# This must be used together with service annotation service.kubernetes.io/topology-mode="auto"
```

File: charts/nginx/values.yaml [hunk 3]
```diff
@@ -85,6 +89,16 @@ electionId: ingress-controller-leader
...
+enableTopologyAwareRouting: false
+
+# -- This configuration disable Nginx Controller Leader Election
+disableLeaderElection: false
```
Comment (danielhoherd) [inline charts/nginx/values.yaml:97]:
```suggestion
disableLeaderElection: false
```

File: charts/nginx/values.yaml [hunk 4]
```diff
@@ -85,6 +89,16 @@ electionId: ingress-controller-leader
...
+
+# -- Duration a leader election is valid before it's getting re-elected, e.g. `15s`, `10m` or `1h`. (Default: 30s)
+electionTTL: ""
```
Comment (danielhoherd) [inline charts/nginx/values.yaml:96]:
```suggestion
electionTTL: "30s"
```

File: charts/nginx/values.yaml (hunk 2 above)
Comment (danielhoherd) [reply charts/nginx/values.yaml]: We don't have a test for `topology-mode`. Do we need one?

File: tests/chart_tests/test_nginx_service.py [hunk 5]
```diff
@@ -235,3 +235,58 @@ def test_nginx_backend_serviceaccount_defaults(self):
...
+    def test_nginx_min_ready_seconds_overrides(self):
+        minReadySeconds = 300
+        doc = render_chart(
+            values={"nginx": {"minReadySeconds": minReadySeconds}},
+            show_only=["charts/nginx/templates/nginx-deployment.yaml"],
+        )[0]
+
+        assert doc["spec"]["minReadySeconds"] == minReadySeconds
+
+    def test_nginx_election_ttl_defaults(self):
```
Comment (danielhoherd) [inline tests/chart_tests/test_nginx_service.py:256]: We can combine these _defaults test cases into a single test case that tests all defaults.

File: charts/nginx/values.yaml (hunk 4 above)
Comment (pgvishnuram) [reply charts/nginx/values.yaml:96]: im okay to remove those

File: tests/chart_tests/test_nginx_service.py (hunk 5 above)
Comment (pgvishnuram) [reply tests/chart_tests/test_nginx_service.py:256]: i was adding these at different timelines i can combine those into one

File: charts/nginx/templates/nginx-deployment.yaml [hunk 6]
```diff
@@ -55,6 +56,12 @@ spec:
             {{- if .Values.enableAnnotationValidations }}
             - --enable-annotation-validation=true
             {{- end }}
+            {{- if .Values.disableLeaderElection }}
+            - --disable-leader-election=true
```
Comment (danielhoherd) [inline charts/nginx/templates/nginx-deployment.yaml:60]: There's no test for this.

**Review by danielhoherd — APPROVED** [review]
We're missing one test case, but other than that it looks great.

**Pull request merged**